crates/api/src/site/mod_log.rs

```diff
@@ -42,69 +53,268 @@ impl Perform for GetModlog {
-          AdminPurgePersonView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgeCommunityView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgePostView::list(conn, mod_person_id, page, limit)?,
-          AdminPurgeCommentView::list(conn, mod_person_id, page, limit)?,
+          match type_ {
+            ModlogActionType::All | ModlogActionType::ModBan => ModBanView::list(
+              conn,
+              mod_person_id,
+              other_person_id,
+              page,
+              limit,
+              hide_modlog_names,
+            )?,
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::ModAdd => ModAddView::list(
+              conn,
+              mod_person_id,
+              other_person_id,
+              page,
+              limit,
+              hide_modlog_names,
+            )?,
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::ModRemoveCommunity => {
+              if other_person_id.is_none() {
+                ModRemoveCommunityView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+              } else {
+                Default::default()
+              }
+            }
+
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::AdminPurgePerson => {
+              if other_person_id.is_none() {
+                AdminPurgePersonView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+              } else {
+                Default::default()
+              }
+            }
+
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::AdminPurgeCommunity => {
+              if other_person_id.is_none() {
+                AdminPurgeCommunityView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+              } else {
+                Default::default()
+              }
+            }
+
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::AdminPurgePost => {
+              if other_person_id.is_none() {
+                AdminPurgePostView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+              } else {
+                Default::default()
+              }
+            }
+
+            _ => Default::default(),
+          },
+          match type_ {
+            ModlogActionType::All | ModlogActionType::AdminPurgeComment => {
+              if other_person_id.is_none() {
+                AdminPurgeCommentView::list(conn, mod_person_id, page, limit, hide_modlog_names)?
+              } else {
+                Default::default()
+              }
+            }
+
+            _ => Default::default(),
+          },
```

Comment:
I think this method still needs some cleanup, but doesnt have to be done in this pr. Also i think it still has the problem that in some cases, old mod log entries are shown before new ones, right?

Reply:
Yeah that's a separate issue that can be solved by doing paging by time-based queries.